aws_lambda_powertools/utilities/data_classes/code_pipeline_job_event.py

```diff
@@ -177,7 +176,7 @@ def user_parameters(self) -> Optional[str]:
         return self.data.action_configuration.configuration.user_parameters
 
     @property
-    def decoded_user_parameters(self) -> Optional[Dict[str, Any]]:
+    def decoded_user_parameters(self) -> Dict[str, Any]:
         """Json Decoded action configuration user parameters"""
         return self.data.action_configuration.configuration.decoded_user_parameters
```

Comment:
We don't need `or {}` because it will return in configurations, right?

Reply:
Correct. `configuration.decoded_user_parameters` already guarantees that at least a `{}` is returned, never `None`. Mypy would also help us here and complain if `configuration.decoded_user_parameters` was annotated to return `Optional[Dict[str, Any]]`, with something like:
```
aws_lambda_powertools/utilities/data_classes/code_pipeline_job_event.py: note: In member "decoded_user_parameters" of class "CodePipelineJobEvent":
aws_lambda_powertools/utilities/data_classes/code_pipeline_job_event.py:181:16: error: Incompatible return value type (got "dict[str, Any] | None", expected "dict[str, Any]")  [return-value]
            return self.data.action_configuration.configuration.decoded_user_parameters
                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```